### PR TITLE
allow to configure an editor language

### PR DESCRIPTION
**Pull request by akosyakov**
_+_ fix displaying bogus current tab size on configuration

TODO:
- [x] check that language servers properly notified

### CODE REVIEW

**Comment by akosyakov** [comment]
It's ready for the review.

**Comment by marechal-p** [comment]
Oh the `Untitled.txt` file is actually a clean up, ok.

File: packages/core/src/browser/quick-open/quick-pick-service.ts [hunk 1]
```diff
@@ -19,8 +19,10 @@ import { QuickOpenItem, QuickOpenMode } from './quick-open-model';
 import { QuickOpenService } from './quick-open-service';
 
 export interface QuickPickItem<T> {
-    label: string,
+    label: string
     value: T
+    description?: string
+    iconClass?: string
```
[the remark this answers is not in the document]
Comment (akosyakov) [reply packages/core/src/browser/quick-open/quick-pick-service.ts]: Because it can break only on statements, in other places it cannot be ambiguous

**Comment by svenefftinge** [comment]
- The coloring doesn't seem to work. I only get coloring for the original language. 
 - When switching between languages, the old diagnostics should be cleaned, otherwise they are sticky. (try switching between TS and python for a *.ts file)
 - Would be good to have the `Auto Detect` as in VS Code option as the top option, so it is easy to switch back.

**Comment by akosyakov** [comment]
> When switching between languages, the old diagnostics should be cleaned, otherwise they are sticky. (try switching between TS and python for a *.ts file)

Looks like a bug in python LS, it gets notified that a document is closed and does nothing. Typescript LS sends empty diagnostics then.

**Comment by akosyakov** [comment]
@svenefftinge all issues have been addressed, also Monaco markers now should properly reflect the state of problem markers and on a language change all problem markers should be cleaned

**Comment by svenefftinge** [comment]
Looks good, it should just filter the list based on what you have typed

**Comment by akosyakov** [comment]
@svenefftinge updated to filter languages, can you have a look once again?

**Pull request merged**